relayer/types/types.go

```diff
@@ -123,6 +113,7 @@ func (m *MessageCache) Add(r *RouteMessage) {
 
 	m.Lock()
 	defer m.Unlock()
+	r.key = key
```

Comment:
try to get key from the route message and then pass it from here....we cannot change the parameter of struct in another struct method...this will create may create unwanted issue...